### PR TITLE
Fix reading CSV in commands

### LINKED ISSUE #468: [BUG] Remove leading zeros from segment names when using `forecast` command
### 🐛 Bug Report

When the `forecast` command reads data from a CSV, it uses the default behavior of the Pandas library, which converts a numeric name to an integer.

Related issue https://github.com/pandas-dev/pandas/issues/31821

### Expected behavior

Storing numeric names as a string

### How To Reproduce

```python3
from pathlib import Path

from etna.commands import forecast

'''
dataset.csv
    =============  ===========  ==========
      timestamp      segment      target
    =============  ===========  ==========
    2020-01-01     000123456         1
    2020-01-02     000123456         2
    2020-01-03     000123456         3
    2020-01-04     000123456         4
    ...
    2020-01-10     000654321        10
    2020-01-11     000654321        20
    =============  ===========  ==========
'''


def main() -> None:
    forecast(
        config_path=Path("config.yaml"),
        forecast_config_path=Path("config.yaml"),
        target_path=Path("dataset.csv"),
        freq="D",
        output_path=Path("result.csv"),
    )


if __name__ == "__main__":
    main()

'''
result.csv
    =============  ===========  ==========
      timestamp      segment      target
    =============  ===========  ==========
    2020-01-01     123456         1
    2020-01-02     123456         2
    2020-01-03     123456         3
    2020-01-04     123456         4
    ...
    2020-01-10     654321        10
    2020-01-11     654321        20
    =============  ===========  ==========
'''
```

### Environment

_No response_

### Additional context

_No response_

### Checklist

- [X] Bug appears at the latest library version

### PR DESCRIPTION
**Pull request by fuglaeff**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #468

### CODE REVIEW

**Comment by fuglaeff** [comment]
I looked at the project code and it seems that these are the only weak points.

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix `IForestOutlierTransform` failed with ignored `target` column ([#460](https://github.com/etna-team/etna/pull/460))
 - Add lower limit for `typing_extension` versions ([#458](https://github.com/etna-team/etna/pull/458))
 - Fix `ModelDecomposeTransform` import without `prophet` module ([#459](https://github.com/etna-team/etna/pull/459))
-- 
+- Fix `backtest` and `forecast` commands converts `segment` to integer ([#468](https://github.com/etna-team/etna/issues/468))
```
Comment (d-a-bunin) [inline CHANGELOG.md:57]: Here we should add link to PR, not to issue.

File: etna/commands/backtest_command.py [hunk 2]
```diff
@@ -82,13 +82,13 @@ def backtest(
         freq_init = freq
         parse_dates = ["timestamp"]
 
-    df_timeseries = pd.read_csv(target_path, parse_dates=parse_dates)
+    df_timeseries = pd.read_csv(target_path, parse_dates=parse_dates, dtype={"segment": str})
```
Comment (d-a-bunin) [inline etna/commands/backtest_command.py:85]: We should probably add some test that checks this behavior. Smth like en example in the issue.

File: etna/commands/forecast_command.py [hunk 3]
```diff
@@ -129,14 +129,13 @@ def forecast(
         freq_init = freq
         parse_dates = ["timestamp"]
 
-    df_timeseries = pd.read_csv(target_path, parse_dates=parse_dates)
-
+    df_timeseries = pd.read_csv(target_path, parse_dates=parse_dates, dtype={"segment": str})
```
Comment (d-a-bunin) [inline etna/commands/forecast_command.py:132]: We should probably add some test that checks this behavior. Smth like en example in the issue.

**Review by d-a-bunin — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -54,7 +54,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix `IForestOutlierTransform` failed with ignored `target` column ([#460](https://github.com/etna-team/etna/pull/460))
 - Add lower limit for `typing_extension` versions ([#458](https://github.com/etna-team/etna/pull/458))
 - Fix `ModelDecomposeTransform` import without `prophet` module ([#459](https://github.com/etna-team/etna/pull/459))
-- 
+- Fix `backtest` and `forecast` commands converts `segment` to integer ([#470](https://github.com/etna-team/etna/pull/470))
```
Comment (d-a-bunin) [inline CHANGELOG.md:57]: May be it is better to rewrite to: 
```
Convert `segment` to string during reading csv in `backtest` and `forecast` commands
```

**Comment by fuglaeff** [comment]
Looked at my mistakes in the tests and fixed them

**Review by d-a-bunin — APPROVED** [review]

**Review by d-a-bunin — APPROVED** [review]

**Pull request merged**